src/monodroid/jni/internal-pinvoke-api.cc

```diff
@@ -0,0 +1,333 @@
+{
+	internal_calls->monodroid_weak_gref_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_weak_gref_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_weak_gref_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_new (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_new (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_delete (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_delete (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_gc_wait_for_bridge_processing (void)
+{
+	internal_calls->monodroid_gc_wait_for_bridge_processing ();
+}
+
+/* !DO NOT REMOVE! Used by Mono BCL */
+MONO_API int
+_monodroid_get_android_api_level (void)
+{
+	return internal_calls->monodroid_get_android_api_level ();
+}
+
+/* Can be called by a native debugger to break the wait on startup */
+MONO_API void
+monodroid_clear_gdb_wait (void)
+{
+	internal_calls->monodroid_clear_gdb_wait ();
+}
+
+MONO_API void*
+_monodroid_get_identity_hash_code (JNIEnv *env, void *v)
+{
+	return internal_calls->monodroid_get_identity_hash_code (env, v);
+}
+
+MONO_API void*
+_monodroid_timezone_get_default_id (void)
+{
+	return internal_calls->monodroid_timezone_get_default_id ();
+}
+
+MONO_API void
+_monodroid_counters_dump (const char *format, ...)
+{
+	va_list args;
+	va_start (args, format);
+	internal_calls->dump_counters (format, args);
+	va_end (args);
+}
+
+/* !DO NOT REMOVE! Used by libgdiplus.so */
+MONO_API int
+_monodroid_get_display_dpi (float *x_dpi, float *y_dpi)
+{
+	return internal_calls->monodroid_get_display_dpi (x_dpi, y_dpi);
+}
+
+MONO_API int
+monodroid_embedded_assemblies_set_assemblies_prefix (const char *prefix)
+{
+	return internal_calls->monodroid_embedded_assemblies_set_assemblies_prefix (prefix);
+}
+
+MONO_API managed_timing_sequence*
+monodroid_timing_start (const char *message)
+{
+	return internal_calls->monodroid_timing_start (message);
+}
+
+MONO_API void
+monodroid_timing_stop (managed_timing_sequence *sequence, const char *message)
+{
+	internal_calls->monodroid_timing_stop (sequence, message);
+}
+
+MONO_API void
+monodroid_strfreev (char **str_array)
+{
+	internal_calls->monodroid_strfreev (str_array);
+}
+
+MONO_API char**
+monodroid_strsplit (const char *str, const char *delimiter, size_t max_tokens)
+{
+	return internal_calls->monodroid_strsplit (str, delimiter, max_tokens);
+}
+
+MONO_API char*
+monodroid_strdup_printf (const char *format, ...)
```

Comment:
Silly Q, but why is this (and others?) exported in the first place?  It's not used from `Mono.Android.dll`, and I can't find any uses of this symbol outside of `src/monodroid` anywhere on GitHub either: https://github.com/search?p=1&q=monodroid_strdup_printf&type=Code

Reply:
I have no clue - I just moved things around, haven't checked if and where from they are still used.

Reply:
A benefit to moving them all into one file is that they're a bit more obvious.  That's nice.